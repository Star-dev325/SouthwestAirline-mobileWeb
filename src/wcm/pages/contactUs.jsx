// @flow
import _ from 'lodash';
import React from 'react';
import { connect } from 'react-redux';
import { retrieveContactUs } from 'src/wcm/actions/wcmActions';
import ContactUsItem from 'src/wcm/components/contactUsItem';
import PageHeader from 'src/shared/components/pageHeader';
import PageFooterWcmSourced from 'src/shared/components/pageFooterWcmSourced';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import withBodyClass from 'src/shared/enhancers/withBodyClass';

import type { WcmFooterType } from 'src/shared/flow-typed/shared.types';

type Props = {
  webContent: {
    title: ?string,
    description: ?string,
    primaryCTAText: ?string,
    primaryCTALink: ?string,
    contentBlock: ?Array<*>
  },
  retrieveContactUsFn: () => void,
  footerLinkRows: Array<WcmFooterType>,
  isWebView: boolean
};

export class ContactUs extends React.Component<Props> {
  componentDidMount() {
    this.props.retrieveContactUsFn();
  }

  render() {
    const {
      webContent: { title, description, primaryCTAText, primaryCTALink, contentBlock = [] },
      footerLinkRows,
      isWebView
    } = this.props;

    return (
      <div>
        <PageHeader data-qa="page-title" className="lineheight20">
          {title}
        </PageHeader>
        <div className="pt6 px4">
          <div className="gray5 my6 lineheight14 large" dangerouslySetInnerHTML={{ __html: description }} />
          <div className="contact-us--call-to-action yellow">
            <a href={primaryCTALink}>{primaryCTAText}</a>
          </div>
          {!!contentBlock &&
            contentBlock.map((block, index) => (
              <div data-qa="contact-us-block" className="mt6" key={index}>
                <h3 className="pt6 pb4 bdt bdgray2 gray5 medium regular caps" data-qa="block-title">
                  {block.blockName}
                </h3>
                <ul className="clearfix m0">
                  {block.items.map((item, idx: string) => (
                    <ContactUsItem key={idx} {...item} />
                  ))}
                </ul>
              </div>
            ))}
        </div>
        {!isWebView && <PageFooterWcmSourced footerLinkRows={footerLinkRows} />}
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  webContent: _.get(state, 'app.wcmContent.contactUs') || {},
  footerLinkRows: _.get(state, 'app.wcmContent.footer.results.footer.content.placement.linkRows', []),
  isWebView: _.get(state, 'app.webView.isWebView')
});

const enhancers = _.flowRight(
  withConnectedReactRouter,
  withBodyClass('bgwhite'),
  connect(mapStateToProps, {
    retrieveContactUsFn: retrieveContactUs
  })
);

export default enhancers(ContactUs);
