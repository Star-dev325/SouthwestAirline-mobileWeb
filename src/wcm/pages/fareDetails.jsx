// @flow
import _ from 'lodash';
import cx from 'classnames';
import PageHeaderWithButtons from 'src/shared/components/pageHeaderWithButtons';
import React from 'react';
import RouterStore from 'src/shared/stores/routerStore';
import StylizedLabel from 'src/shared/components/stylizedLabel';
import withBodyClass from 'src/shared/enhancers/withBodyClass';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import { connect } from 'react-redux';
import { getFareDetails } from 'src/wcm/selectors/fareDetailsSelectors';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';

import type { Location } from 'react-router';
import type { Push } from 'src/shared/flow-typed/shared.types';

type PageContent = {
  title?: string,
  headings?: Array<*>,
  categories?: { [string]: Array<*> }
};

type Props = {
  pageContent: PageContent,
  goBack: () => void,
  push: Push,
  location: Location
};

export const FareDetails = ({
  pageContent: { title, headings = [], categories: categoriesObject },
  goBack,
  push
}: Props) => {
  const _transitionToAvailablePage = () => {
    if (RouterStore.getPrevState()) {
      goBack();
    } else {
      push(getNormalizedRoute({ routeName: 'index' }));
    }
  };

  const _renderPageHeadings = () => (
    <div className="fare-details--headings_wrapper">
      {_.map(headings, (heading, index: number) => {
        const dividerStyles = heading.textColor && {
          borderColor: heading.textColor
        };

        return (
          <div
            className="fare-details--fare"
            data-qa="fare-details-page-heading"
            key={index}
            style={{
              background: heading.backgroundColor,
              color: heading.textColor || 'white'
            }}
          >
            <div className="fare-details--fare-name_wrapper">
              <span className="fare-details--fare-name">
                <StylizedLabel value={heading.stylizedFareName} defaultText={heading.fareName} />
              </span>
            </div>
            <div className="fare-details--fare-points">
              <div className="fare-details--fare-points_amount" style={dividerStyles}>
                <span>{heading.amountOfPoints}</span>
              </div>
              <div className="fare-details--fare-points_per-dollar">
                <span>{heading.pointsPerDollar}</span>
              </div>
            </div>
          </div>
        );
      })}
    </div>
  );

  const _renderCategoriesAndValues = () => (
    <div className="flex flex-column fare-details--categories-and-values">
      {_.map(categoriesObject, (categories: Array<*>, key: string) => {
        const values = key.split('--');
        const categoryTitle = values[0];
        const index = parseInt(values[1]);
        const description = _.get(categories, '0.description', '');

        return (
          <div
            key={index}
            data-qa="fare-details-page-category"
            className={cx('flex flex-cross-center flex-cross-baseline py6 p12 pdkblue', {
              bgwhite: index % 2 === 0,
              bggray2: index % 2 !== 0
            })}
          >
            <div className="flex-column center flex5">
              {_.map(categories, (valuePerFare, idx: number) =>
                (valuePerFare.value ? (
                  <span key={idx} style={{ background: valuePerFare.color }} className="active-circle p4 m2" />
                ) : (
                  <span key={idx} className="p4 m2" />
                ))
              )}
            </div>
            <div className="flex-column flex7 pr5 fare-details--attribute">
              <div className="large" dangerouslySetInnerHTML={{ __html: categoryTitle }} />
              <div
                className="fare-details--attribute-description py3"
                dangerouslySetInnerHTML={{ __html: description }}
              />
            </div>
          </div>
        );
      })}
    </div>
  );

  return (
    <div className="fare-details">
      <PageHeaderWithButtons
        rightButtons={[{ name: 'Done', onClick: _transitionToAvailablePage }]}
        title={title}
        dataQa="fare-details-page-title"
      />
      {_renderPageHeadings()}
      {_renderCategoriesAndValues()}
    </div>
  );
};

const mapStateToProps = (state) => ({
  pageContent: getFareDetails()(state)
});

const enhancers = _.flowRight(withConnectedReactRouter, connect(mapStateToProps), withBodyClass('fare-details-page'));

export default enhancers(FareDetails);
