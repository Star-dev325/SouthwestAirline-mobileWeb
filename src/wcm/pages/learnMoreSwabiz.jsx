// @flow
import _ from 'lodash';
import React from 'react';
import { connect } from 'react-redux';
import i18n from '@swa-ui/locale';
import { retrieveLearnMoreSwabiz } from 'src/wcm/actions/wcmActions';
import RouterStore from 'src/shared/stores/routerStore';
import WcmOverlayModal from 'src/wcm/components/wcmOverlayModal';
import withBodyClass from 'src/shared/enhancers/withBodyClass';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import { raiseSatelliteEvent } from 'src/shared/analytics/helpers/analyticsEventHelper';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';

import type { OverlayType, Push } from 'src/shared/flow-typed/shared.types';

type Props = {
  goBack: () => void,
  isWebView: boolean,
  learnMoreSwabizContent?: OverlayType,
  push: Push,
  wcmFetchActionFn: () => void
};

export class LearnMoreSwabiz extends React.Component<Props> {
  componentDidMount() {
    this.props.wcmFetchActionFn();
    raiseSatelliteEvent('what is swabiz');
  }
  _transitionToAirBookingPage = () => {
    const { goBack, push } = this.props;

    RouterStore.getPrevState() ? goBack() : push(getNormalizedRoute({ routeName: 'index' }));
  };

  render() {
    const { learnMoreSwabizContent, isWebView } = this.props;

    return (
      <WcmOverlayModal
        {...{
          overlay: learnMoreSwabizContent,
          onDone: this._transitionToAirBookingPage,
          doneLabel: i18n('AIR_BOOKING__CORPORATE_BOOKING__DONE'),
          isWebView
        }}
      />
    );
  }
}

const mapStateToProps = (state) => ({
  isWebView: _.get(state, 'app.webView.isWebView'),
  learnMoreSwabizContent: _.get(state, 'app.wcmContent.learnMoreSwabiz.overlay', null)
});

const mapDispatchToProps = {
  wcmFetchActionFn: retrieveLearnMoreSwabiz
};

const enhancers = _.flowRight(
  withConnectedReactRouter,
  withBodyClass('hide-header'),
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(LearnMoreSwabiz);
