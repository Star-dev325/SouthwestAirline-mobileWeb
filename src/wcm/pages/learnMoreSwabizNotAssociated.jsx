// @flow
import _ from 'lodash';
import React from 'react';
import { connect } from 'react-redux';
import WcmOverlayModal from 'src/wcm/components/wcmOverlayModal';
import { retrieveLearnMoreSwabizNotAssociated } from 'src/wcm/actions/wcmActions';
import RouterStore from 'src/shared/stores/routerStore';
import withBodyClass from 'src/shared/enhancers/withBodyClass';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import { raiseSatelliteEvent } from 'src/shared/analytics/helpers/analyticsEventHelper';
import i18n from '@swa-ui/locale';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';

import type { OverlayType, Push } from 'src/shared/flow-typed/shared.types';

type Props = {
  goBack: () => void,
  isWebView: boolean,
  learnMoreSwabizContent?: OverlayType,
  push: Push,
  wcmFetchActionFn: () => void
};

export class LearnMoreSwabizNotAssociated extends React.Component<Props> {
  componentDidMount() {
    this.props.wcmFetchActionFn();
    raiseSatelliteEvent('no company id associated');
  }

  _transitionToAirBookingPage = () => {
    const { goBack, push } = this.props;

    RouterStore.getPrevState() ? goBack() : push(getNormalizedRoute({ routeName: 'index' }));
  };

  render() {
    const { learnMoreSwabizContent, isWebView } = this.props;

    return (
      <WcmOverlayModal
        {...{
          overlay: learnMoreSwabizContent,
          onDone: this._transitionToAirBookingPage,
          doneLabel: i18n('AIR_BOOKING__CORPORATE_BOOKING__DONE'),
          isWebView
        }}
      />
    );
  }
}

const mapStateToProps = (state) => ({
  isWebView: _.get(state, 'app.webView.isWebView'),
  learnMoreSwabizContent: _.get(state, 'app.wcmContent.learnMoreSwabizNotAssociated.overlay', null)
});

const mapDispatchToProps = {
  wcmFetchActionFn: retrieveLearnMoreSwabizNotAssociated
};

const enhancers = _.flowRight(
  withConnectedReactRouter,
  withBodyClass('hide-header'),
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(LearnMoreSwabizNotAssociated);
