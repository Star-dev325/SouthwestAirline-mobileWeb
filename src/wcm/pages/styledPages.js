import wcmStyledPage from 'src/wcm/pages/wcmStyledPage';
import {
  retrieveInTheAir,
  retrieveFlyingSouthwest,
  retrieveAtTheAirport,
  retrieveBoardingThePlane,
  retrieveAboutRapidRewards
} from 'src/wcm/actions/wcmActions';

export const InTheAirPage = wcmStyledPage(retrieveInTheAir, 'inTheAir');

export const FlyingSouthwestPage = wcmStyledPage(retrieveFlyingSouthwest, 'flyingSouthwest');

export const AtTheAirportPage = wcmStyledPage(retrieveAtTheAirport, 'atTheAirport');

export const BoardingThePlanePage = wcmStyledPage(retrieveBoardingThePlane, 'boardingThePlane');

export const AboutRapidRewardsPage = wcmStyledPage(retrieveAboutRapidRewards, 'aboutRapidRewards');
