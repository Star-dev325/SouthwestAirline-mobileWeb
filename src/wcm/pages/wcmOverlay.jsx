// @flow
import _ from 'lodash';
import React from 'react';
import { connect } from 'react-redux';
import withHideGlobalHeader from 'src/shared/enhancers/withHideGlobalHeader';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import withBodyClass from 'src/shared/enhancers/withBodyClass';
import Overlay from 'src/wcm/components/overlay';
import PageHeaderWithButtons from 'src/shared/components/pageHeaderWithButtons';

import type { OverlayType } from 'src/shared/flow-typed/shared.types';

type Props = {
  overlay: OverlayType,
  wcmFetchActionFn: () => void,
  showBackButton?: boolean
};

const WcmOverlay = (wcmFetchAction: () => *, nodeName: string) => {
  class WcmOverlayComponent extends React.Component<Props> {
    componentDidMount() {
      this.props.wcmFetchActionFn();
    }

    render() {
      const { overlay, showBackButton } = this.props;
      const { title, body } = overlay;

      return (
        <div>
          {body && (
            <div>
              <PageHeaderWithButtons titleInCenter title={title} showBackButton={showBackButton} />
              <div data-qa="wcm-overlay">
                <Overlay body={body} />
              </div>
            </div>
          )}
        </div>
      );
    }
  }

  WcmOverlayComponent.displayName = `${_.upperFirst(nodeName)}Overlay`;

  const mapStateToProps = (state) => ({
    overlay: _.get(state, `app.wcmContent.${nodeName}.overlay`) || {}
  });

  const mapDispatchToProps = {
    wcmFetchActionFn: wcmFetchAction
  };

  const enhancers = _.flowRight(
    withConnectedReactRouter,
    withHideGlobalHeader,
    withBodyClass('hide-header'),
    connect(mapStateToProps, mapDispatchToProps)
  );

  return enhancers(WcmOverlayComponent);
};

export default WcmOverlay;
