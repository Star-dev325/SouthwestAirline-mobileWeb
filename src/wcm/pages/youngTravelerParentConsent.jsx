// @flow
import i18n from '@swa-ui/locale';
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import withBodyClass from 'src/shared/enhancers/withBodyClass';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import { flowRight, get } from 'src/shared/helpers/jsUtils';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';
import RouterStore from 'src/shared/stores/routerStore';
import { retrieveYoungTravelerParentConsent } from 'src/wcm/actions/wcmActions';
import WcmOverlayModal from 'src/wcm/components/wcmOverlayModal';
import { defaultYoungTravelerParentConsent } from 'src/wcm/constants/wcmFallbackConstants';

import type { OverlayType, Push } from 'src/shared/flow-typed/shared.types';

type Props = {
  goBack: () => void,
  isWebView: boolean,
  overlay?: OverlayType,
  push: Push,
  retrieveYoungTravelerParentConsentFn: () => void
};

export const YoungTravelerParentConsent = ({
  goBack,
  isWebView,
  overlay,
  push,
  retrieveYoungTravelerParentConsentFn
}: Props) => {
  useEffect(() => {
    retrieveYoungTravelerParentConsentFn();
  }, []);
  
  const transitionToAirBookingPage = () => {
    RouterStore.getPrevState() ? goBack() : push(getNormalizedRoute({ routeName: 'index' }));
  };

  return (
    <WcmOverlayModal
      {...{
        doneLabel: i18n('AIR_BOOKING__CORPORATE_BOOKING__DONE'),
        isWebView,
        onDone: transitionToAirBookingPage,
        overlay
      }}
    />
  );
};

const mapStateToProps = (state) => ({
  isWebView: get(state, 'app.webView.isWebView'),
  overlay: get(state, 'app.wcmContent.youngTravelerParentConsent.overlay', defaultYoungTravelerParentConsent)
});

const mapDispatchToProps = {
  retrieveYoungTravelerParentConsentFn: retrieveYoungTravelerParentConsent
};

const enhancers = flowRight(
  withConnectedReactRouter,
  withBodyClass('hide-header'),
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(YoungTravelerParentConsent);
