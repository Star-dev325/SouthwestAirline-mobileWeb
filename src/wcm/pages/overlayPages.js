import {
  retrieveTaxesAndFees,
  retrieveHazardousMaterials,
  retrieveBaggageRestrictions,
  retrieveFareRules,
  retrievePrivacyPolicy,
  retrieveTermsAndConditions,
  retrieveFormsOfPayment,
  retrieveCancellationPolicy,
  retrieveCarriageContract,
  retrieveCheckinAndRefund,
  retrieveCarLimitOfLiability,
  retrieveStandbyPolicies,
  retrieveEarlyBirdIntroduction,
  retrieveFareRulesForFareType,
  retrieveSpecialAssistance,
  retrieveTravelFundsTermsConditions
} from 'src/wcm/actions/wcmActions';
import wcmOverlay from 'src/wcm/pages/wcmOverlay';

export const TaxesAndFeesOverlay = wcmOverlay(retrieveTaxesAndFees, 'taxesAndFees');

export const HazardousMaterialsOverlay = wcmOverlay(retrieveHazardousMaterials, 'hazardousMaterials');

export const BaggageRestrictionsOverlay = wcmOverlay(retrieveBaggageRestrictions, 'baggageRestrictions');

export const FareRulesOverlay = wcmOverlay(retrieveFareRules, 'fareRules');

export const PrivacyPolicyOverlay = wcmOverlay(retrievePrivacyPolicy, 'privacyPolicy');

export const TermsAndConditionsOverlay = wcmOverlay(retrieveTermsAndConditions, 'termsAndConditions');

export const FormsOfPaymentOverlay = wcmOverlay(retrieveFormsOfPayment, 'formsOfPayment');

export const CancellationPolicyOverlay = wcmOverlay(retrieveCancellationPolicy, 'cancellationPolicy');

export const CarriageContractOverlay = wcmOverlay(retrieveCarriageContract, 'carriageContract');

export const CheckinAndRefundOverlay = wcmOverlay(retrieveCheckinAndRefund, 'checkinAndRefund');

export const CarLimitOfLiabilityOverlay = wcmOverlay(retrieveCarLimitOfLiability, 'carLimitOfLiability');

export const StandbyPoliciesOverlay = wcmOverlay(retrieveStandbyPolicies, 'standbyPolicies');

export const EarlyBirdIntroductionOverlay = wcmOverlay(retrieveEarlyBirdIntroduction, 'earlyBirdIntroduction');

export const SpecialAssistanceOverlay = wcmOverlay(retrieveSpecialAssistance, 'specialAssistance');

export const TravelFundsTermsConditionsOverlay = wcmOverlay(
  retrieveTravelFundsTermsConditions,
  'travelFundsTermsConditions'
);

export const FareRulesForFareTypeOverlays = (props) => {
  const fareType = props.match.params.fareType.split('-').join('_');

  return wcmOverlay(() => retrieveFareRulesForFareType(fareType), 'fareRulesForFareType');
};
