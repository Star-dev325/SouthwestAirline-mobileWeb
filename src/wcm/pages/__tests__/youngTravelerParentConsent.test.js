import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { airBookingRoutes } from 'src/airBooking/constants/airBookingRoutes';
import * as AppSelector from 'src/shared/selectors/appSelector';
import RouterStore from 'src/shared/stores/routerStore';
import { defaultYoungTravelerParentConsent } from 'src/wcm/constants/wcmFallbackConstants';
import { YoungTravelerParentConsent } from 'src/wcm/pages/youngTravelerParentConsent';
import { createMockedFormStore } from 'test/unit/helpers/formTestUtils';

describe('YoungTravelerParentConsent', () => {
  const TITLE = 'Young Traveler Parent Consent';
  let goBackStub;
  let pushStub;
  let retrieveYoungTravelerParentConsentFnStub;

  beforeEach(() => {
    goBackStub = jest.fn();
    pushStub = jest.fn();
    retrieveYoungTravelerParentConsentFnStub = jest.fn();
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  describe('when loading wcm content', () => {
    it('should use default content if wcm call fails', () => {
      retrieveYoungTravelerParentConsentFnStub.mockReturnValueOnce(Promise.reject());

      const { container } = createComponent({
        overlay: defaultYoungTravelerParentConsent
      });

      expect(container).toMatchSnapshot();
    });

    it('should call wcm content endpoint', () => {
      createComponent();

      expect(retrieveYoungTravelerParentConsentFnStub).toHaveBeenCalled();
    });

    it('should use push if there is no history', () => {
      jest.spyOn(AppSelector, 'getCurrentAppFlow').mockReturnValueOnce('air/booking');

      const { container } = createComponent();

      fireEvent.click(container.querySelector('button'));

      expect(pushStub).toHaveBeenCalledWith(airBookingRoutes['index'].canonicalPath);
    });

    it('should use push if there is history', () => {
      jest.spyOn(RouterStore, 'getPrevState').mockReturnValue({ routeState: 'route' });

      const { container } = createComponent();

      fireEvent.click(container.querySelector('button'));

      expect(goBackStub).toHaveBeenCalled();
    });
  });

  const createComponent = (props = {}, state = {}) => {
    const defaultProps = {
      goBack: goBackStub,
      isWebView: false,
      overlay: {
        body: [],
        title: TITLE
      },
      push: pushStub,
      retrieveYoungTravelerParentConsentFn: retrieveYoungTravelerParentConsentFnStub
    };
    const combinedProps = {
      ...defaultProps,
      ...props
    };

    return render(
      <Provider store={createMockedFormStore(state)}>
        <YoungTravelerParentConsent {...combinedProps} ref={props.ref} />
      </Provider>
    );
  };
});
