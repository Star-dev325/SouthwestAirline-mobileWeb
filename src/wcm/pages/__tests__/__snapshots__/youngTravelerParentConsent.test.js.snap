// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoungTravelerParentConsent when loading wcm content should use default content if wcm call fails 1`] = `
<div>
  <div
    class="wcm-overlay-modal"
  >
    <div>
      <div
        class="page-header action-bar"
        id=""
      >
        <ul
          class="action-bar-buttons action-bar--left-buttons"
        />
        <div
          class="action-bar--container"
        >
          <div
            class="action-bar--title"
          >
            <span
              class="action-bar--main-title"
            >
              Young Traveler parent consent
            </span>
            <span
              class="action-bar--sub-title"
            />
          </div>
        </div>
        <ul
          class="action-bar-buttons action-bar--right-buttons"
        >
          <li
            class="action-bar-buttons--item"
          >
            <button
              class="button"
              type="button"
            >
              AIR_BOOKING__CORPORATE_BOOKING__DONE
            </button>
          </li>
        </ul>
      </div>
      <div />
    </div>
    <div
      class="wcm-content"
    >
      <div>
        <p
          class="p3"
        >
          <strong>
            • 
          </strong>
          Southwest® considers children ages 12 through 17 traveling alone to be Young Travelers, not Unaccompanied Minors. Accordingly, Southwest® Unaccompanied Minor procedures do not apply to ages 12 through 17.
        </p>
        

        <p
          class="p3"
        >
          <strong>
            • 
          </strong>
          Parents/Guardians are responsible for ensuring a Young Traveler arrives for check-in and boarding in accordance with each airport's suggested arrival time. The amount of time to park, check luggage, pass through the security checkpoint, and board the flight varies based on each airport's size, configuration, and travel season.
        </p>
        

        <p
          class="p3"
        >
          <strong>
            • 
          </strong>
          Your child has sufficient maturity and capability to travel alone (this includes, but is not limited to, checking in, passing through the security checkpoint, boarding, deplaning, and claiming luggage), without adult supervision or assistance from Southwest® Employees, on all scheduled service: nonstop, same-plane service with intermediate stops, and connecting service that requires your child to change planes.
        </p>
        

        <p
          class="p3"
        >
          <strong>
            • 
          </strong>
          Parents/Guardians wishing to escort a Young Traveler to/from the gate must obtain a Non-Passenger Escort (NPE) from the ticket counter in order to pass through the security checkpoint. Each airport may impose restrictions on NPEs being issued based on guidance from the local Transportation Security Administration directives.
        </p>
        

        <p
          class="p3"
        >
          <strong>
            • 
          </strong>
          Southwest® does not monitor Young Travelers during travel. Accordingly, each Young Traveler must have sufficient maturity and capability to request assistance from a Southwest® Employee if needed and the means to contact a parent/guardian if there is a travel disruption.
        </p>
        

        <p
          class="p3"
        >
          <strong>
            • 
          </strong>
          Southwest Airlines® reserves the right not to allow travel on flights that may be delayed, diverted, or cancelled due to weather or other operational disruptions.
        </p>
        

        <p
          class="p3"
        >
          <strong>
            • 
          </strong>
          Southwest Airlines® does not provide Young Traveler service to/from international destinations.
        </p>
        

        <p
          class="p3"
        >
          <strong>
            • 
          </strong>
          Southwest Airlines® does not assume responsibility for any Young Traveler's actions. THE PURCHASER AGREES TO INDEMNIFY, HOLD HARMLESS, AND FULLY RELEASE SOUTHWEST AIRLINES® FOR ANY AND ALL INJURY OR DAMAGES TO PROPERTY OR PERSONS CAUSED BY OR INCURRED AS A RESULT OF A YOUNG TRAVELER'S ACTIONS, INCLUDING INJURIES TO THE YOUNG TRAVELER CAUSED BY HIS OR HER OWN NEGLIGENCE.
        </p>
      </div>
    </div>
  </div>
</div>
`;
